crates/fuel-core/src/database/contracts.rs

```diff
@@ -16,19 +16,114 @@ use fuel_core_storage::{
         ContractsRawCode,
     },
     ContractsAssetKey,
+    Error as StorageError,
+    Mappable,
     Result as StorageResult,
     StorageAsRef,
+    StorageInspect,
+    StorageMutate,
 };
-use fuel_core_types::fuel_types::{
-    AssetId,
-    Bytes32,
-    ContractId,
-    Word,
+use fuel_core_types::{
+    fuel_tx::Contract,
+    fuel_types::{
+        AssetId,
+        Bytes32,
+        ContractId,
+        Word,
+    },
 };
+use serde::Deserializer;
+use std::borrow::Cow;
 
-impl DatabaseColumn for ContractsRawCode {
-    fn column() -> Column {
-        Column::ContractsRawCode
+struct OptimizedContract {
+    bytes: Vec<u8>,
+}
+
+impl From<OptimizedContract> for Contract {
+    fn from(value: OptimizedContract) -> Self {
+        value.bytes.into()
+    }
+}
+
+impl<'de> serde::Deserialize<'de> for OptimizedContract {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        struct BytesVisitor;
+
+        impl<'a> serde::de::Visitor<'a> for BytesVisitor {
+            type Value = Vec<u8>;
+
+            fn expecting(
+                &self,
+                formatter: &mut core::fmt::Formatter,
+            ) -> core::fmt::Result {
+                formatter.write_str("a borrowed byte array")
+            }
+
+            fn visit_borrowed_bytes<E>(self, v: &'a [u8]) -> Result<Self::Value, E>
+            where
+                E: serde::de::Error,
+            {
+                Ok(v.to_vec())
+            }
+        }
+
+        let bytes = deserializer.deserialize_bytes(BytesVisitor {})?;
+        Ok(Self { bytes })
+    }
+}
```

Comment:
Can this be replace by [serde_bytes](https://docs.rs/serde_bytes/latest/serde_bytes/)?

Reply:
We can, I only want to have fewer dependencies and the code that do exactly what we want. But we plan to remove it in the master, so I'm okay to use `serder_bytes`